### Homework 8

**Chapter 5**
12.) Reconsider the space docking problem of Example 5.2.

**Example 5.2**
Reconsider the docking problem of Example 4.3, and assume now that it takes $5$ seconds to make the control adjustments, and another $10$ seconds until we can return from other tasks to observe the velocity indicator once again. Under these conditions, will our strategy for matching velocities be successful?

**Example 4.3**. Astronauts in training are required to practice a docking maneuver under manual control. As a part of this maneuver, it is required to bring an orbiting spacecraft to rest relative to another orbiting craft. The hand controls provide for variable acceleration and deceleration, and there is a device on board that measures the rate of closing between the two vehicles. The following strategy has been proposed for bringing the craft to rest. First, look at the
closing velocity. If it is zero, we are done. Otherwise, remember the closing velocity and look at the acceleration control. Move the acceleration control so that it is opposite to the closing velocity (i.e., if closing velocity is positive, we slow down, and we speed up if it is negative) and proportional in magnitude (i.e., we brake twice as hard if we find ourselves closing twice as fast). After a time, look at the closing velocity again and repeat the procedure. Under what circumstances will this strategy be effective?

(From the textbook):

Let $v_n$ denote the closing velocity observed at time $t_n$, the time of the nth observation. Let $\Delta v_n = v_{n+1} - v_n$ denote the change in closing velocity as a result of our adjustments. The time interval divides into two parts: the time it takes to adjust the velocity controls and the time between adjustments and the next observation of the velocity indicator. Write $\Delta t_n = c_n + w_n$, where $c_n$ is the time to adjust the controls and $w_n$ is the waiting time until the next observation. 

Let $a_n$ denote the acceleration setting after the nth adjustment. Elementary physics yields $\Delta v_n = a_{n-1}c_n + a_n w_n.$ The control law says to set acceleration proportional to $(-v_n)$; hence $a_n = -kv_n$. These results are summarized below.

$$t_{n+1} = t_n + c_n + w_n$$
$$v_{n+1} = v_n + a_{n−1}c_n + a_nw_n$$
$$a_n = −kv_n$$
$$c_n > 0$$
$$w_n ≥ 0$$



(b) Find the eigenvectors associated with the eigenvalues
$$\lambda = \frac{4 \pm \sqrt{6}}{10}$$

that were calculated in the text?


   Start by defining all the important variables for calculating acceleration velocity and our matrix $F$. We let
   
   $$x_1(n) = v_n \, \, \, \text{and} \, \, \, x_2(n) = v_{n-1}.$$
   
So a successful docking procedure involves $F(x_1, x_2) \rightarrow (0,0)$. From $F$ we derive our matrix as the coefficents of $G = x + F(x)$ so $g_1(x_1, x_2) = 0.8x_1 - 0.1x_2$ and $g_2(x_1, x_2) = x_1$. This gives us the matrix 

$$ A = 
\left(\begin{array}{cc} 
0.8 & -0.1\\ 
1 & 0
\end{array}\right). $$

The textbook already calculated the eigenvalues of this matrix as 
$$\lambda_{1,2} = \frac{4 \pm \sqrt{6}}{10}.$$

We calculate the eigenvectors by solving the system of equations that arise from $Av_1 = \lambda_1 v_1$ and $Av_2 = \lambda_2 v_2$ where $v_1,v_2$ are the eigenvectors associated with $\lambda_1$ and $\lambda_2$ respectively. So the sake of brevity I omit the steps to find the vectors, they come out to be

$$v_1 = 
\left(\begin{array}{cc}
\lambda_1 \\
1
\end{array}\right).$$

$$v_2 = 
\left(\begin{array}{cc}
\lambda_2 \\
1
\end{array}\right).$$


We will also define the eigenvectors here are *eigenvector_1* and *eigenvector_2*

In [10]:
#function for calculating the acceleration 
acceleration(velocity) = -0.02 * velocity

#acc_0 and acc_1 are the current (acc_1) and the previous (acc_0) accelerations
velocity(velocity_0,acc_0,acc_1) = velocity_0 + 5*acc_0 + 10*acc_1

#
F(x1,x2) = [-0.02*10*x1 - 0.02*5*x2, x1-x2]

#Two eigenvalues
λ₁= (4 + sqrt(6))/10
λ₂= (4 - sqrt(6))/10

#Two eigenvectors
eigenvector_1 = (λ₁,1)
eigenvector_2 = (λ₂,1)

(0.15505102572168222, 1)

**(c)** Calculate the rate at which closing velocity decreases (% per minute)
if we start at an eigenvector.

We are using the same model from HW6 that calculates the velocity as the docking procedure occurs.

In [7]:
#x and y are the initial velocities of the past two steps
function spacemodel(x,y)
    #initialize the arrays that hold the 
    x1 = zeros(100,1)
    x2 = zeros(100,1)
    
    #sets two initial values
    x1[1] = x
    x2[1] = y
    
    println("Velocity of Spaceship at time 0: ( ",x," )")
    
    #only interate 4 times since each time is 15 seconds
    for j in 1:4
        x,y = F(x1[j],x2[j])
        
        x = x + x1[j]
        y = y + x2[j]
        
        x1[j+1] = x
        x2[j+1] = y
        println("Velocity of Spaceship at time ", j,": ( ",x," )")
    end
end

spacemodel (generic function with 1 method)

In [12]:
spacemodel(eigenvector_1[1], eigenvector_1[2])

Velocity of Spaceship at time 0: ( 0.6449489742783178 )
Velocity of Spaceship at time 1: ( 0.4159591794226542 )
Velocity of Spaceship at time 2: ( 0.26827244611029155 )
Velocity of Spaceship at time 3: ( 0.17302203894596782 )
Velocity of Spaceship at time 4: ( 0.1115903865457451 )


In [14]:
spacemodel(eigenvector_2[1], eigenvector_2[2])

Velocity of Spaceship at time 0: ( 0.15505102572168222 )
Velocity of Spaceship at time 1: ( 0.024040820577345773 )
Velocity of Spaceship at time 2: ( 0.0037275538897083987 )
Velocity of Spaceship at time 3: ( 0.0005779610540321413 )
Velocity of Spaceship at time 4: ( 8.961345425487308e-5 )


Calculating the velocity decrease at eigenvector_1 we have 

$$\frac{0.6449 - 0.1116}{0.6449} = 83\%.$$

Velocity decrease at eigenvector_2 we have 

$$\frac{0.1551 - (8.961 * 10^{-5})}{0.1551} = 99\%.$$

This implies that the modl congerves much quicker along the second eigenvector.